### DISCUSSION TITLE
Feature Request: Umbraco Cloud fault tolerance

### DISCUSSION THREAD

**Discussion by gmanrodt** [umbraco/Umbraco.Cloud.Issues#898] [Ideas]
### Issue description

This topic covers a lot of ground, but I was asked to post this as a "Feature Request", so here goes:

- It is the expectation of all of our clients that we be able to deploy new features and functionality to their website without any downtime. Umbraco Cloud forces an app pool recycle on every deployment.
- There is no way to automate/schedule backups of Umbraco Cloud environments. SQL backups can be manually executed, then downloaded from the portal, but that is insufficient for an company of any real size. Additionally, waiting 24 hours for Umbraco Support to provide a backup is an insufficient SLA.
- It is not possible for environments to be load-balanced
- There are no disaster recovery options
- No region pairing (what if the US East Azure data center has connectivity issues? It has happened before...)
- No availability zones (what if Microsoft needs to patch the VMs where the Umbraco sites live? Or move the storage location?)
- No backups of any kind for the media library assets, unless you contact Umbraco Support. Umbraco indicated they had backups of the media items from BLOB storage, but again, 24 hours is too long to wait

**Comment by BenDuguid-MRM** [comment]
So much all of this.
